idseq/uploader.py

```diff
@@ -36,6 +40,30 @@ def parts(self):
         else:
             return [self.path]
 
+def detect_files(path, level=1):
+    wildcards = "/*" * level
+    return [f for f in glob.glob(path + wildcards) if re.search(INPUT_REGEX, f) and os.stat(f).st_size > 0]
+
+def detect_samples(path):
+    files_level1 = detect_files(path, level=1)
+    files_level2 = detect_files(path, level=2)
+    if not files_level1 and not files_level2:
+        print "No fastq/fasta files found"
+        raise Exception()
+    samples2files = {}
+    if len(files_level1) >= len(files_level2):
```

Comment:
Hm this is a bit arbitrary?

Reply:
should prolly also document how the files are supposed to be organized? 
we should probably try detect files_level1 first, if we don't find any good pair try level 2.

Reply:
Makes sense. But sometimes people have unclean folders, with perhaps a stray fastq file at the top level but then many more actual files in the subfolders?

Reply:
Well. we should ask them to keep things clean there otherwise we will have to deal with many many edge cases.

Reply:
ok, fixed